lib/Fhp/FinTs.php

```diff
@@ -283,22 +283,30 @@ public function execute(BaseAction $action)
             throw new \RuntimeException('Need to login (DialogInitialization) before executing other actions');
         }
 
-        // Let the BaseAction implementation build its request segments.
-        $requestSegments = $action->createRequest($this->bpd, $this->upd);
-        $requestSegments = is_array($requestSegments) ? $requestSegments : [$requestSegments];
-        if (count($requestSegments) === 0) {
-            return; // No request needed.
-        }
-        $this->checkPaginationToken($action, $requestSegments);
-
-        // Construct the full request message.
-        $message = MessageBuilder::create()->add($requestSegments); // This fills in the segment numbers.
-        if (!($this->getSelectedTanMode() instanceof NoPsd2TanMode)) {
-            $needTanForSegment = $this->bpd->tanRequiredForRequest($requestSegments);
-            if ($needTanForSegment !== null) {
-                $message->add(HKTANv6::createProzessvariante2Step1(
-                    $this->requireTanMode(), $this->selectedTanMedium, $needTanForSegment));
+        if (($message = $action->getMessage()) === null) {
```

Comment:
Ich verstehe das `[0]` unten noch nicht ganz. Aber wenn das nicht im Weg steht, wäre es vielleicht möglich, eine `final BaseAction::getNextRequest()` Funktion zu haben, die sich intern um `$this->message` kümmert, ggf. `createRequest()` aufruft und einfach immer eine Message (nie null) zurückgibt.

Reply:
Ja das klingt gut, ich wollte im ersten Anlauf nicht so viel umstrukturieren, aber kann es mal in die Richtung probieren.